src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -50,6 +51,10 @@ public class DefaultFunctions {
 	private static String str(double n) {
 		return StringUtils.toString(n, 4);
 	}
+	private static final String DEFAULT_NUMBER_FORMAT_STRING = "###,###";
+	private static final DecimalFormat DEFAULT_NUMBER_FORMAT = new DecimalFormat(DEFAULT_NUMBER_FORMAT_STRING);
+	private static final ThreadLocal<DecimalFormat> NUMBER_FORMAT = ThreadLocal.withInitial(DecimalFormat::new);
```

Comment:
Can I ask what the point of this is? Maybe I'm being dense but I don't see A) what an empty decimalformat is doing and B) why it's in a threadlocal.

Reply:
See [comment](https://github.com/SkriptLang/Skript/pull/4663#discussion_r1554422551) :)

Reply:
Is it not okay to just use one 😦